src/async-wrap.cc

```diff
@@ -407,6 +407,47 @@ static void DisablePromiseHook(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+class DestroyParam {
+ public:
+  double asyncId;
+  v8::Persistent<Object> target;
+  v8::Persistent<Object> propBag;
+};
+
+
+void AsyncWrap::WeakCallback(
+  const v8::WeakCallbackInfo<DestroyParam>& info
+) {
+  HandleScope scope(info.GetIsolate());
+
+  Environment* env = Environment::GetCurrent(info.GetIsolate());
+  DestroyParam* p = info.GetParameter();
+  Local<Object> propBag = PersistentToLocal(info.GetIsolate(), p->propBag);
+
+  Local<Value> val = propBag->Get(env->destroyed_string());
+  if (!val->BooleanValue()) {
```

Comment:
`if (val->IsFalse())`? Otherwise you’ll need to switch to the variant of `BooleanValue()` that is not deprecated (i.e. the one that takes a `Local<Context> context` argument)